### PR TITLE
Depracation warning for parameter names not being valid python identifiers

### PR DESCRIPTION
**Pull request by quaquel**
See #274 for details. This just turns the exception into a deprecation warning in preparation for the 2.5 release

### CODE REVIEW

File: ema_workbench/em_framework/util.py [hunk 1]
```diff
@@ -62,6 +62,13 @@ def variable_name(self, name):
             name = [name]
         self._variable_name = name
 
+    def __init__(self, name):
+        if not name.isidentifier():
+            DeprecationWarning(
+                f"'{name}' is not a valid Python identifier. In future versions of the workbench, names must be valid python identifiers"
```
Comment (EwoutH) [inline ema_workbench/em_framework/util.py:68]:
```suggestion
                f"'{name}' is not a valid Python identifier. Starting from version 3.0 of the EMAworkbench, names must be valid python identifiers"
```

**Comment by EwoutH** [comment]
Globally, looks good!

Ideally, clean up the commits into:
- The actual code changes
- The warning added
- Test changed
- All the model / example changes (so all the conversions, underscores)
- Files changed

Otherwise squashing on merge would also be fine. I would just make sure it doesn't cause major conflicts with #274.

**Comment by quaquel** [comment]
How would I manage reorganizing the commits because I agree that a clean history would be a convenient.

**Comment by EwoutH** [comment]
In my process I use [GitHub desktop](https://desktop.github.com/). You can drag and drop commits to reorder them. Then you squash the ones that work on the same feature (for example squashing all the testing commits to one commit). See [Reordering commits in GitHub Desktop](https://docs.github.com/en/desktop/managing-commits/reordering-commits-in-github-desktop).

If you're more of a CLI person: [7.6 Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History).

**Pull request merged**